src/diffusers/pipelines/controlnet/pipeline_controlnet.py

```diff
@@ -837,6 +843,20 @@ def __call__(
                 second element is a list of `bool`s indicating whether the corresponding generated image contains
                 "not-safe-for-work" (nsfw) content.
         """
+        # We do this first to capture the "true" call values. If we do this at a later point in time,
+        # we cannot ensure that the true call values weren't changed during the process.
+        # We update the `generator` later, though as we define a new generator in case it was passed as `None`.
+        if generator is None:
+            seed = random.randint(0, MAX_SEED)
+            generator = torch.manual_seed(seed)
+
+        workflow = None
+        if return_workflow:
+            signature = inspect.signature(self.__call__)
+            argument_names = [param.name for param in signature.parameters.values()]
+            call_arg_values = inspect.getargvalues(inspect.currentframe()).locals
+            workflow = populate_workflow_from_pipeline(argument_names, call_arg_values, self)
```

Comment:
I'm pretty sure we can move all of this into `PipelineMixin` with a:

```py
def populate_workflow_from_pipeline(self, ....):
             signature = inspect.signature(self.__call__)
            argument_names = [param.name for param in signature.parameters.values()]
            call_arg_values = # get local from previous frame 
```

You should be able to get the local from the previous frame via: https://stackoverflow.com/questions/3711184/how-to-use-inspect-to-get-the-callers-info-from-callee-in-python

Can you maybe try that?

Reply:
Done: https://github.com/huggingface/diffusers/pull/4825/commits/474df7a6b1629e29898ceedd173b2043877c3b4d